src/main/java/org/broadinstitute/consent/http/models/DacDecisionMetrics.java

```diff
@@ -202,16 +187,39 @@ public Integer getPercentAgreementAlgorithm() {
     return percentAgreementAlgorithm;
   }
 
-  private void setPercentAgreementAlgorithm(Integer percentAgreementAlgorithm) {
-    this.percentAgreementAlgorithm = percentAgreementAlgorithm;
+  private void setPercentAgreementAlgorithm(List<DarDecisionMetrics> completedDarMetrics) {
+    if (!completedDarMetrics.isEmpty()) {
+      List<DarDecisionMetrics> agreementMetricsDenominator =
+        completedDarMetrics.stream()
+          .filter(m -> Objects.nonNull(m.getAlgorithmDecision()))
+          .collect(Collectors.toList());
+
+      List<DarDecisionMetrics> agreementMetricsNumerator =
+        agreementMetricsDenominator.stream()
+          .filter(m -> m.getAlgorithmDecision().equalsIgnoreCase(m.getDacDecision()))
+          .collect(Collectors.toList());
+
+      int percentAgreement = (int) (((double) agreementMetricsNumerator.size() / (double) completedDarMetrics.size()) * 100);
+      this.percentAgreementAlgorithm = percentAgreement;
+    }
   }
 
   public Integer getPercentSRPAccurate() {
     return percentSRPAccurate;
   }
 
-  private void setPercentSRPAccurate(Integer percentSRPAccurate) {
-    this.percentSRPAccurate = percentSRPAccurate;
+  private void setPercentSRPAccurate(List<DarDecisionMetrics> completedDarMetrics) {
+    List<DarDecisionMetrics> srpMetricsDenominator =
+      completedDarMetrics.stream()
+        .filter(m -> Objects.nonNull(m.getSrpDecision()))
+        .collect(Collectors.toList());
+
+    List<DarDecisionMetrics> srpMetricsNumerator =
+      srpMetricsDenominator.stream()
+        .filter(m -> m.getSrpDecision().equalsIgnoreCase("yes"))
+        .collect(Collectors.toList());
+    int percentSrp = (int) (((double) srpMetricsNumerator.size() / (double) srpMetricsDenominator.size()) * 100);
```

Comment:
One suggestion for refactoring. Since we're calculating the percentages multiple times (3 times in this PR), it is probably a good idea to extract that out to a private method that takes two `int`s and does the same thing in terms of casting to doubles (nice catch with `long`->`double`, by the way) and then returning the value in `int` form. That way we're not leaving it up to each method to figure out the right implementation via copy/paste.